### ISSUE TITLE
GDB ReadMemoryCommand fails when dumping memory sections that are larger than page size or span across pages

### ISSUE THREAD

**Issue by kartben** [renode/renode#171]
When using GDB's `dump binary` command or VS Code's Cortex-Debug "View Memory" command to dump/read a memory section that is larger than page size or across two pages, the command fails due to a recent change in ReadMemoryCommand preventing reads that span multiple pages. 

```
dump binary memory result.bin a_large_buffer a_large_buffer+65536
```

**Comment by mateusz-holenko** [comment]
@kartben This should be fixed with eb04ace46589f708f944d34bbc5af9a3fbc77f2d, could you check if this works for you?

**Comment by PiotrZierhoffer** [comment]
Closing, as this should work just fine right now.
